zenoh/src/api/session.rs

```diff
@@ -824,16 +827,25 @@ impl Session {
     }
 }
 
+/// Like a [`Session`], but not closed on drop
+pub(crate) struct SessionClone(ManuallyDrop<Session>);
```

Comment:
It's unclear to me the need of such `SessionClone`.  What real problem is it solved here?
Since `SessionClone` contains clones of `Runtime`, `State`, etc. how the state is cleaned up?
How is this change robust to internal changes, e.g. `Runtime` will required to be properly cleaned? If that happens, it seems we will be leaking memory or not freeing resources.

Reply:
Yes, that was a mistake of mine.